### PR TITLE
Ignore custom property sets

### PR DESCRIPTION
**Pull request by jeddy3**
Borrows a conditional from [`postcss-minify-selectors`](https://github.com/ben-eb/postcss-minify-selectors/blob/master/src/index.js#L14) to ignore custom property sets in the `selector-no-*` and `selector-*-pattern` rules.

### CODE REVIEW

File: src/utils/cssRuleHasSelectorEndingWithColon.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Check whether a rule has a selector ending in a colon
+ *
+ * @param {Rule} postcss rule node
+ * @return {boolean} If `true`, the rule has a selector ending in a colon
+ */
+export default function (rule) {
+  const selector = rule.raws.selector && rule.raws.selector.raw || rule.selector
```
Comment (davidtheclark) [inline src/utils/cssRuleHasSelectorEndingWithColon.js]: I like cleaning up this kind of thing with lodash's `_.get()`: `const selector = _.get(rule, 'raws.selector.raw', rule.selector)` But that's up to you, just an idea.

**Comment by davidtheclark** [comment]
Feel free to merge. Does this close #626?

**Comment by jeddy3** [comment]
> Does this close #626?

It does. The auto-close message got [buried a bit](https://github.com/stylelint/stylelint/commit/d31e7bc15250f094bff96361652b3529ae65d214).

File: src/utils/cssRuleHasSelectorEndingWithColon.js (hunk 1 above)
Comment (jeddy3) [inline src/utils/cssRuleHasSelectorEndingWithColon.js]: Coolio. I'll make the change. I saw you making use of lodash's functions to clean things up e.g. using `includes`. I'm not very familiar with all of lodash's functions though and so if you spot any other opportunities please let me know.

Comment (davidtheclark) [inline src/utils/cssRuleHasSelectorEndingWithColon.js]: Yeah, I'm getting very tired of writing the `indexOf() !== -1` pattern :)

**Pull request merged**